sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientCodeSnippets.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.data.tables.implementation.models.TableServiceErrorException;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class TableClientCodeSnippets {
+
+    public static void methods() {
+
+        //create a tableServiceClient
+        TableServiceClient tableServiceClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildClient();
+
+        //create TableClient
+        TableClient tableClient = new TableClientBuilder()
+            .connectionString("connectionString")
+            .tableName("OfficeSupplies")
+            .buildClient();
+
+
+        //create a table
+        try {
+            tableServiceClient.createTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            //use azure core errors? based on
+            System.out.println("Create Table Unsuccessful. Error: " + e);
+        }
+
+        //delete  table
+        try {
+            tableServiceClient.deleteTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            System.out.println("Delete Table Unsuccessful. Error: " + e);
+        }
+
+        //query tables
+        String selectString = "$selectString= TableName eq 'OfficeSupplies'";
+
+        try {
+            List<AzureTable> responseTables = tableServiceClient.queryTables(selectString);
+        } catch (HttpResponseException e) {
+            System.out.println("Table Query Unsuccessful. Error: " + e);
+        }
+
+
+        //insert entity
+        String tableName = "OfficeSupplies";
```

Comment:
A `TableClient` already requires a `tableName` in order to create one, which implies to me that a given instance of `TableClient` only operates on the table that was specified when it was created. So if that's the case, we should avoid making the customer pass the table name again here.